library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -0,0 +1,143 @@
+                inputStream = new FileInputStream(new File(originalFileLocation));
+                byte[] bytes = new byte[BUFFER_SIZE];
+
+                // read the v1 file
+                int readLast = 0;
+                while (readLast != -1) {
+                    readLast = inputStream.read(bytes);
+                    if (readLast != 0 && readLast != -1) {
+                        // write the v1 file to the v2 location
+                        internalFilePersistence.write(bytes, 0, readLast);
+                        bytes = new byte[BUFFER_SIZE];
+                    }
+                }
+            } catch (IOException e) {
+                Log.e(getClass().getSimpleName(), e.getMessage());
+                e.printStackTrace();
+            } finally {
+                try {
+                    internalFilePersistence.close();
+                    if (inputStream != null) {
+                        inputStream.close();
+                    }
+                } catch (IOException e) {
+                    Log.e(getClass().getSimpleName(), e.getMessage());
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private void migrateV1DataToV2Database(List<Migration> migrations) {
+        for (Migration migration : migrations) {
+            Batch batch = migration.batch();
+            List<FileSize> fileSizes = migration.fileSizes();
+            List<String> originalFileLocations = migration.originalFileLocations();
+
+            migrateV1DataToV2Database(batch, fileSizes, originalFileLocations);
+        }
+    }
+
+    private void migrateV1DataToV2Database(Batch batch, List<FileSize> fileSizes, List<String> originalFileLocations) {
+        downloadsPersistence.startTransaction();
+
+        DownloadBatchTitle downloadBatchTitle = new LiteDownloadBatchTitle(batch.getTitle());
+        DownloadsBatchPersisted persistedBatch = new LiteDownloadsBatchPersisted(downloadBatchTitle, batch.getDownloadBatchId(), Status.DOWNLOADED);
+        downloadsPersistence.persistBatch(persistedBatch);
+
+        for (int i = 0; i < originalFileLocations.size(); i++) {
+            String url = batch.getFileUrls().get(i);
+            FileName fileName = LiteFileName.from(batch, url);
+            FilePath filePath = FilePathCreator.create(fileName.name());
+            DownloadFileId downloadFileId = DownloadFileId.from(batch);
+            DownloadsFilePersisted persistedFile = new LiteDownloadsFilePersisted(
+                    batch.getDownloadBatchId(),
+                    downloadFileId,
+                    fileName,
+                    filePath,
+                    fileSizes.get(i).totalSize(),
+                    url,
+                    FilePersistenceType.INTERNAL
+            );
+            downloadsPersistence.persistFile(persistedFile);
+        }
+
+        downloadsPersistence.transactionSuccess();
+        downloadsPersistence.endTransaction();
```

Comment:
I guess the catch and abort scenario is just not added yet since you mentioned happy case in the description